src/scripts/main.js

```diff
@@ -1,3 +1,81 @@
 'use strict';
 
-// write code here
+const tBody = document.querySelector('tbody');
+const minLimit = 2;
+const maxLimit = 10;
+const fields = {
+  rows: tBody.rows.length,
+  cols: tBody.rows[0].cells.length,
+};
+
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+fields.switchRowButtons = () => {
+  appendRow.disabled = fields.rows >= maxLimit;
+  removeRow.disabled = fields.rows <= minLimit;
+};
+
+fields.switchColumnButtons = () => {
+  appendColumn.disabled = fields.cols >= maxLimit;
+  removeColumn.disabled = fields.cols <= minLimit;
+};
+
+const makeAppendRow = () => {
+  if (appendRow.disabled) {
+    return;
+  }
+
+  const newRow = tBody.insertRow();
+
+  for (let i = 0; i < fields.cols; i++) {
+    newRow.insertCell(-1);
+  }
+
+  fields.rows++;
+  fields.switchRowButtons();
+};
+
+const makeRemoveRow = () => {
+  if (removeRow.disabled) {
+    return;
+  }
+
+  tBody.deleteRow(-1);
```

Comment:
Consider checking if tBody is not null and has rows before calling deleteRow() to avoid potential runtime errors.